client/gutenberg/editor/components/post-trash-link/index.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { restorePost, trashPost } from 'state/posts/actions';
+import { navigate } from 'state/ui/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import getPostTypeTrashUrl from 'state/selectors/get-post-type-trash-url';
+
+export class PostTrashLink extends Component {
+	onClick = () => {
+		const { postId, siteId, trashUrl } = this.props;
```

Comment:
Any reason for not using the object destructuring as well for getting `trashPost` and `navigate`?

```suggestion
		const { navigate, postId, siteId, trashPost, trashUrl } = this.props;
```

Reply:
It's just because they have the same names as their imports, and so they would then be `already declared in the upper scope. eslint(no-shadow)` 🙂